charts/spire/values.yaml

```diff
@@ -27,9 +27,32 @@ spire-agent:
   clusterName: *clusterName
   trustDomain: *trustDomain
 
+upstream:
+  # -- enable upstream csi driver and agent for use with nested spire.
+  enabled: false
+
+upstream-spire-agent:
+  nameOverride: agent-upstream
+  bundleConfigMap: spire-bundle-upstream
+
+  clusterName: *clusterName
+  trustDomain: *trustDomain
+
+  socketPath: /run/spire/agent-sockets-upstream/spire-agent.sock
+  serviceAccount:
+    name: spire-agent-upstream
+  healthChecks:
+    port: 9981
```

Comment:
```suggestion
upstream:
  # -- enable upstream csi driver and agent for use with nested spire.
  enabled: false

  spire-agent:
    nameOverride: agent-upstream
    bundleConfigMap: spire-bundle-upstream

    clusterName: *clusterName
    trustDomain: *trustDomain

    socketPath: /run/spire/agent-sockets-upstream/spire-agent.sock
    serviceAccount:
      name: spire-agent-upstream
    healthChecks:
      port: 9981

  spiffe-csi-driver:
    pluginName: upstream.csi.spiffe.io
    agentSocketPath: /run/spire/agent-sockets-upstream/spire-agent.sock
    healthChecks:
      port: 9810
```

Reply:
I'd love to do the api like that, but there is a problem. You can only nest charts one level per nested chart.

so to have it like that, we'd need an intermediate chart named 'upstream' under charts/spire/charts/
I think it would work, but would look just a little funny on the filesystem.

Do you have any issues with that arrangement? If not, I'll prototype it out.